csharp/server/ElevatorServer.cs

```diff
@@ -0,0 +1,133 @@
+            elevatorServer.Run();
+
+            Console.WriteLine("A simple webserver. Press a key to quit.");
+            Console.ReadKey();
+            elevatorServer.Stop();
+        }
+
+        private readonly HttpListener _listener = new HttpListener();
+        private readonly ElevatorEngine _elevatorEngine = new ElevatorEngine();
+
+        public ElevatorServer(string root)
+        {
+            if (string.IsNullOrEmpty(root))
+            {
+                throw new ArgumentException("root");
+            }
+
+            _listener.Prefixes.Add(root);
+            _listener.Start();
+        }
+
+        public void Run()
+        {
+            ThreadPool.QueueUserWorkItem((o) =>
+            {
+                Console.WriteLine("Webserver running...");
+                try
+                {
+                    while (_listener.IsListening)
+                    {
+                        ThreadPool.QueueUserWorkItem((c) =>
+                        {
+                            var ctx = c as HttpListenerContext;
+                            try
+
+                            {
+                                switch (ctx.Request.Url.Segments[1].Replace("/", ""))
+                                {
+                                    case "reset":
+                                    {
+                                        _elevatorEngine.Reset(ctx.Request.QueryString.Get("cause"));
+                                        HttpResponseOk(ctx);
+                                        break;
+                                    }
+                                    case "call":
+                                    {
+                                        Direction direction =
+                                            (Direction) Enum.Parse(typeof(Direction), ctx.Request.QueryString.Get("to"));
+                                        _elevatorEngine.Call(
+                                            Int32.Parse(ctx.Request.QueryString.Get("atFloor")),
+                                            direction
+                                        );
+                                        HttpResponseOk(ctx);
+                                        break;
+                                    }
+                                    case "go":
+                                    {
+                                        _elevatorEngine.Go(
+                                            Int32.Parse(ctx.Request.QueryString.Get("floorToGo"))
+                                        );
+                                        HttpResponseOk(ctx);
+                                        break;
+                                    }
+                                    case "nextCommand":
+                                    {
+                                        _elevatorEngine.NextCommand();
+                                        HttpResponseOk(ctx);
```

Comment:
This request needs a response body filled with next command. For now, command returned back from `_elevatorEngine.NextCommand()` is lost.